### PR TITLE
added unique id to related; fixed collection

### PR DESCRIPTION
**Pull request by vzhenia**
* added unique string to related field (for FB5)
* fixed false positives on ads identification (sponsored text was identified if empty string was passed in or text that contained letters in necessary order, for example: 
FOUND========== Sponsa in Elon Musk performs a song about everyday life of the astronaut
FOUND========== Sponsrad in Account switcherAnother person has unread notifications.
FOUND========== Sponsored in Advance your career with online courses in programming, data science, artificial intelligence, digital marketing, and more. Built with industry leaders)
* fixed collection of rationales (FB5)
* added some handlers to variables stored in local storage

### CODE REVIEW

File: src/daemon/collector/content/collect.js [hunk 1]
```diff
@@ -750,20 +747,25 @@ window.addEventListener("message", function(event) {
...
-  let keys = Object.keys(rq).sort();
-  let visited = [];
-  let res = {};
-  if (keys.length) {
-    const adIds = keys.map(k => rq[k].adId);
-    for (let i=0; i<keys.length; i++) {
-      if (!visited.includes(adIds[i])) {
-        res[keys[i]] = rq[keys[i]];
-        visited.push(adIds[i]);
+  if (rq) {
```
[the remark this answers is not in the document]
Comment (vzhenia) [reply src/daemon/collector/content/collect.js:750]: rq is a queue to store data for rationale requests

File: src/daemon/collector/content/collect.js [hunk 2]
```diff
@@ -262,53 +256,52 @@ function hasSillySponsored(haystackText) {
...
-    if (sponsoredText.indexOf(lst[i].text) > -1
-      && (lst[i].getAttribute('class') && lst[i].getAttribute('class').indexOf(non_ad) < 0)
-      && !isScrolledIntoView(lst[i], newStyle) ){
-        // console.log(lst[i])
-        // console.log('******filter Front Ads**HIDDEN********');
-    }
-
-    if (hasSillySponsored(lst[i].text) && isScrolledIntoView(lst[i], newStyle)) {
+    // empty strings should not trigger false positives
+    if (lst[i].text && hasSillySponsored(lst[i].text) && isScrolledIntoView(lst[i], layoutStyle)) {
```
Comment (skinofstars) [inline src/daemon/collector/content/collect.js:286]: This `hasSillySponsored` is returning too many false positives. Please can we remove this check for this release?

Comment (vzhenia) [reply src/daemon/collector/content/collect.js:286]: ok

File: src/daemon/collector/content/collect.js [hunk 3]
```diff
@@ -490,51 +487,46 @@ function findFrontAdsWithHiddenLettersSiblings(){
...
+  // console.log('!!!!!raw_ad ------ collected', layoutStyle, frontAd)
+  // console.log('!!!!!raw_ad ------ parent_id', layoutStyle, parent_id)
 
   // ----- Temporarily collect ads even if rationales are not there ----- //
   const container = $(raw_ad);
   let fbStoryId;
-  if (newStyle) {
-    fbStoryId = parent_id;
+  if (layoutStyle === "FB5") {
+    fbStoryId = generateRelatedField(parent_id);
```
Comment (skinofstars) [inline src/daemon/collector/content/collect.js:525]: what is this id going to look like?

Comment (vzhenia) [reply src/daemon/collector/content/collect.js:525]: somewhat like this "jsc_c_4_317c3fd7-3e83-47f6-87e1-e40f8f70eb50"

Comment (skinofstars) [reply src/daemon/collector/content/collect.js:525]: Does the `jsc_c_4_` bit still have purpose?

**Pull request merged**